src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -60,16 +90,33 @@
     // Contains boundary MultiPolygons
     private final CountryBoundaryMap countryBoundaryMap;
 
-    private final RawAtlasChangeSet afterSlicingLinesAndPoints = new RawAtlasChangeSet();
-    private final RawAtlasChangeSet afterSlicingRelations = new RawAtlasChangeSet();
+    // Keep track of changes made during Point/Line and Relation slicing
+    private final RawAtlasSimpleChangeSet slicedPointAndLineChanges = new RawAtlasSimpleChangeSet();
+    private final RawAtlasRelationChangeSet slicedRelationChanges = new RawAtlasRelationChangeSet();
 
     // Mapping between Coordinate and created Temporary Point identifiers. This is to avoid
     // duplicate points at the same locations and to allow fast lookup to construct new lines
     // requiring the temporary point as a Line shape point
-    private final Map<Coordinate, Long> newPointCoordinates = new HashMap<>();
+    private final CoordinateToNewPointMapping newPointCoordinates = new CoordinateToNewPointMapping();
 
     private final RawAtlasSlicingStatistic statistics = new RawAtlasSlicingStatistic(logger);
 
+    /**
+     * JTS has trouble dealing with high-precision double values. For this reason, we round all
+     * coordinates to 7 degrees of precision. See {@link Coordinate} java doc for more detailed
+     * explanation of scaling.
+     *
```

Comment:
We are storing all Latitude/Longitude values with dm7 precision, to be consistent with the rest of the code base. The class above, that references 6 degrees of precision, is only used to store a mapping between Coordinate to Point identifier. This is only used for quick lookup to verify we don't have duplicate points, otherwise everything is dm7.